src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,210 @@
+import uk.gov.hmcts.reform.domain.CaseData;
+import uk.gov.hmcts.reform.domain.OldApplicant;
+import uk.gov.hmcts.reform.domain.common.Address;
+import uk.gov.hmcts.reform.domain.common.ApplicantParty;
+import uk.gov.hmcts.reform.domain.common.EmailAddress;
+import uk.gov.hmcts.reform.domain.common.TelephoneNumber;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class ApplicantDataMigrationServiceTest {
+    private final String APPLICANT = "Applicant name";
+    private final String EMAIL = "email@email.com";
+    private final String MOBILE = "07778987656";
+    private final String COUNTY = "County";
+    private final String COUNTRY = "Country";
+    private final String POSTCODE = "BT17NT";
+    private final String POST_TOWN = "Ballymena";
+    private final String ADDRESS_LINE_1 = "ADDRESS_LINE_1";
+    private final String ADDRESS_LINE_2 = "ADDRESS_LINE_2";
+    private final String ADDRESS_LINE_3 = "ADDRESS_LINE_3";
+    private final String JOB_TITLE = "JobTitle";
+    private final String TELEPHONE = "02825674837";
+    private final String PERSON_TO_CONTACT = "Person to contact";
+    private final String PBA_NUMBER = "PBA123456";
+
+    @Autowired
+    private ApplicantDataMigrationService service;
+
+    @Test
+    void whenOldStructureExistsAcceptsShouldReturnTrue() {
+        Map<String, Object> data = new HashMap<>();
+
+        OldApplicant applicant = createOldApplicant();
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(true);
+    }
+
+    @Test
+    void whenDataIsNullAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(null)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenNewStructureDoesNotExistAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(ImmutableMap.of("data", "someData"))
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenOldApplicantStructureIsMigratedShouldReturnNewApplicantStructure() {
+        Map<String, Object> data = new HashMap<>();
+
+        data.put("applicant", createOldApplicant());
+
+        CaseData migratedData = service.migrate(data);
+
+        ApplicantParty actualValue = migratedData.getApplicants().get(0).getValue().getParty();
+        ApplicantParty expectedValue = createNewApplicant().getParty();
+
+        assertThat(migratedData.getApplicant()).isNull();
+
+        assertThat(actualValue.getPartyId()).isNotNull();
+        assertThat(actualValue.getPartyType()).isNull();
```

Comment:
No party type is checked / set. We should check for exact value as we know what it should be.

REQ